core/arch/arm/kernel/secure_partition.c

```diff
@@ -570,6 +570,49 @@ static TEE_Result handle_fdt_dev_regions(struct sp_ctx *ctx, void *fdt)
 	return TEE_SUCCESS;
 }
 
+static TEE_Result handle_fdt_endpoint_id(struct sp_session *s,
+					 const void * const fdt)
+{
+	uint32_t endpoint_id = 0;
+
+	if (!sp_dt_get_u32(fdt, 0, "id", &endpoint_id)) {
+		struct sp_session *a_session = sp_get_session(endpoint_id);
+
+		if (endpoint_id <= SPMC_ENDPOINT_ID)
+			return TEE_ERROR_BAD_FORMAT;
+
+		/* Switch the id's If the id was already assigned. */
+		if (a_session)
+			a_session->endpoint_id = s->endpoint_id;
+
+		s->endpoint_id = endpoint_id;
+	}
+	return TEE_SUCCESS;
+}
+
+static TEE_Result check_endpoint_ids(void)
+{
+	const struct sp_image *sp = NULL;
+
+	for_each_secure_partition(sp) {
+		uint32_t fdt_id = 0;
+
+		if (!sp_dt_get_u32(sp->fdt, 0, "id", &fdt_id)) {
+			TEE_Result res = TEE_SUCCESS;
+			uint32_t sp_id = 0;
+
+			res = sp_find_session_id(&sp->image.uuid, &sp_id);
+			if (res)
+				return res;
+			if (sp_id != fdt_id) {
+				EMSG("SP: invalid endpoint id");
```

Comment:
This could probably be a DMSG instead. It wouldn't hurt to print the problematic UUID and ids too.

Reply:
Ack